solidity/dashboard/src/components/Tooltip.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState, useRef } from "react"
+import * as Icons from "./Icons"
+
+const Tooltip = ({
+  triggerComponent: TriggerComponent,
+  title,
+  content,
+  children,
+  direction = "bottom",
+  icon: IconComponent = Icons.Tooltip,
+  simple = false,
+  delay = 300,
+}) => {
+  const timeout = useRef(null)
+  const [active, setActive] = useState(false)
+
+  useEffect(() => {
+    return () => {
+      if (timeout.current) {
+        clearTimeout(timeout.current)
+      }
+    }
+  })
+
+  const showTooltip = () => {
+    timeout.current = setTimeout(() => {
+      setActive(true)
+    }, delay)
+  }
+
+  const hideTooltip = () => {
+    clearInterval(timeout.current)
+    setActive(false)
+  }
+
+  return (
+    <div className={`tooltip${simple ? "--simple" : ""}--${direction}`}>
+      <div
+        className="tooltip__trigger"
+        onMouseEnter={showTooltip}
+        onMouseLeave={hideTooltip}
+      >
+        <TriggerComponent />
+      </div>
+      {active && (
+        <div
+          className="tooltip__content-wrapper"
+          onMouseEnter={() => setActive(true)}
+          onMouseLeave={hideTooltip}
+        >
+          {children ? (
+            children
+          ) : (
+            <>
+              <Tooltip.Header text={title} icon={IconComponent} />
+              <Tooltip.Divider />
```

Comment:
The overall structure here seems a little more complicated that it might need to be for what we're trying to achieve… Nothing urgent, but would be nice to revisit at some point from the perspective of how we can use semantic markup and targeted styling to potentially collapse some of the layers a little bit.

Reply:
The main idea was to achieve maximum flexibility in case to create custom tooltip components via passing `children` and all necessary "logic" is inside tooltip component. But if skip `children` and pass `title`, `IconComponent` and `content` props we render the default tooltip component from [KEEP Design System](https://www.figma.com/file/ARAMFLMOPal0JewjvlZGhX/Keep-Design-System?node-id=1764%3A3)

Reply:
Do we use that flexibility right now, or is it more hypothetical?

Reply:
Probably more hypothetical, but we have some tooltips with redirect button and some without. Also, I think is better to have this flexibility instead of passing props like: `withBtn`, `topBtnProps`, `bottomBtnProps` etc. in case we would add a new type of tooltip.
Maybe we should just leave `{children}` and creates a something like `ResourceTooltip` that would render this:

![obraz](https://user-images.githubusercontent.com/57687279/100254100-2fba0800-2f42-11eb-81a1-8ee1e75cb02a.png)

Reply:
That could be an interesting alternative to look into, definitely!